src/coreclr/jit/assertionprop.cpp

```diff
@@ -3177,33 +3177,78 @@ GenTree* Compiler::optVNConstantPropOnTree(BasicBlock* block, GenTree* tree)
 
     if (conValTree != nullptr)
     {
-        if (tree->OperIs(GT_LCL_VAR))
+        if (tree->OperIs(GT_LCL_VAR) && conValTree->OperIs(GT_CNS_VEC, GT_CNS_DBL, GT_CNS_INT))
         {
-            if (conValTree->OperIs(GT_CNS_VEC) && !conValTree->IsVectorZero() && !conValTree->IsVectorAllBitsSet())
+            bool keepPropagating = false;
+
+            if (conValTree->OperIs(GT_CNS_VEC))
+            {
+                if (conValTree->IsVectorZero() || conValTree->IsVectorAllBitsSet())
```

Comment:
For Arm64, values fit in `fmov` are also "cheap". I believe we have a helper for detecting them.

Reply:
Added a helper, thanks